### PR TITLE
.gitlab-ci cleanup | Resolves issue #4226

### PR DESCRIPTION
**Pull request by AyushyaChitransh**
This pull request refers to Issue of .gitlab-ci clean-up #4226

### CODE REVIEW

**Comment by AyushyaChitransh** [comment]
The PR needs a review from @General-Beck or someone from @paritytech/ci

**Comment by debris** [comment]
lgtm, @jesuscript can you also review this pr?

**Comment by General-Beck** [comment]
I think here I will make more optimizations. and this script does not work in windows. tomorrow I will change the part and check

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -500,50 +248,7 @@ windows:
...
-    - aws s3api put-object --bucket %S3_BUCKET% --key %CI_BUILD_REF_NAME%/x86_64-pc-windows-msvc/parity.zip.md5 --body target\release\parity.zip.md5
-    - aws s3api put-object --bucket %S3_BUCKET% --key %CI_BUILD_REF_NAME%/x86_64-pc-windows-msvc/InstallParity.exe --body nsis\InstallParity.exe
-    - aws s3api put-object --bucket %S3_BUCKET% --key %CI_BUILD_REF_NAME%/x86_64-pc-windows-msvc/InstallParity.exe.md5 --body nsis\InstallParity.exe.md5
-    - aws s3api put-object --bucket %S3_BUCKET% --key %CI_BUILD_REF_NAME%/x86_64-pc-windows-msvc/win-installer.zip --body nsis\win-installer.zip
-    - aws s3api put-object --bucket %S3_BUCKET% --key %CI_BUILD_REF_NAME%/x86_64-pc-windows-msvc/win-installer.zip.md5 --body nsis\win-installer.zip.md5
-    - curl --data "commit=%CI_BUILD_REF%&sha3=%SHA3%&filename=parity.exe&secret=%RELEASES_SECRET%" http://update.parity.io:1337/push-build/%CI_BUILD_REF_NAME%/%PLATFORM%
-    - curl --data "commit=%CI_BUILD_REF%&sha3=%SHA3%&filename=parity.exe&secret=%RELEASES_SECRET%" http://update.parity.io:1338/push-build/%CI_BUILD_REF_NAME%/%PLATFORM%
+    - ./scripts/gitlab-ci.sh windows
```
Comment (jacogr) [inline .gitlab-ci.yml]: Not sure how this part will run on Windows, it doesn't support bash by default.

File: scripts/gitlab-ci.sh [hunk 2]
```diff
@@ -0,0 +1,451 @@
+linux-stable(){
```
Comment (jacogr) [inline scripts/gitlab-ci.sh]: Don't we need a `#!/bin/sh` in here? (Additionally, just make sure the script in +x - not sure how to check for review)

File: scripts/gitlab-ci.sh [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+      aws s3 rm --recursive s3://$S3_BUCKET/$CI_BUILD_REF_NAME/$PLATFORM
+      aws s3api put-object --bucket $S3_BUCKET --key $CI_BUILD_REF_NAME/$PLATFORM/parity --body target/release/parity
+      aws s3api put-object --bucket $S3_BUCKET --key $CI_BUILD_REF_NAME/$PLATFORM/parity.md5 --body parity.md5
+      aws s3api put-object --bucket $S3_BUCKET --key $CI_BUILD_REF_NAME/$PLATFORM/"parity-"$VER"-macos-installer.pkg" --body "parity-"$VER"-macos-installer.pkg"
+      aws s3api put-object --bucket $S3_BUCKET --key $CI_BUILD_REF_NAME/$PLATFORM/"parity-"$VER"-macos-installer.pkg.md5" --body "parity-"$VER"-macos-installer.pkg.md5"
+      curl --data "commit=$CI_BUILD_REF&sha3=$SHA3&filename=parity&secret=$RELEASES_SECRET" http://update.parity.io:1337/push-build/$CI_BUILD_REF_NAME/$PLATFORM
+      curl --data "commit=$CI_BUILD_REF&sha3=$SHA3&filename=parity&secret=$RELEASES_SECRET" http://update.parity.io:1338/push-build/$CI_BUILD_REF_NAME/$PLATFORM
+}
+
+windows(){
```
Comment (jacogr) [inline scripts/gitlab-ci.sh]: Rather move to `gitlab-ci.bat` and be certain it is for Windows. (See previous comment)

File: scripts/gitlab-ci.sh (hunk 2 above)
Comment (AyushyaChitransh) [reply scripts/gitlab-ci.sh]: Not necessarily, but its good to have `#!/bin/sh`

**Comment by AyushyaChitransh** [comment]
There have been some updates to fix previously mentioned problems. And the script works on windows platform also now.

**Comment by debris** [comment]
@General-Beck @paritytech/ci please review again

**Comment by tomusdrw** [comment]
@paritytech/ci please review

**Comment by 5chdn** [comment]
It's in f400a7e1718dddd77f4d60f65c91134dbbfb3a4a now - thanks :)

**Pull request closed without merging**